src/main/scala/mimir/adaptive/SchemaMatching.scala

```diff
@@ -0,0 +1,169 @@
+        ) => {
+          val modelsByColAndType =
+            constructor(
+              db, 
+              s"$viewName:$modelCategory", 
+              Left(config.query), 
+              Right(targetSchema)
+            ).toSeq.map {
+              case (col, (model, idx)) => (col, (model, idx, Seq[Expression]()))
+            }
+
+          (modelCategory, modelsByColAndType)
+        }
+      }
+
+    val (
+      candidateModels: Map[String,Seq[(String,Int,Seq[Expression],String)]],
+      modelEntities: Seq[Model]
+    ) = 
+      LensUtils.extractModelsByColumn(modelsByType)
+
+    // Sanity check...
+    targetSchema.map(_._1).foreach( target => {
+      if(!candidateModels.contains(target)){ 
+        throw new Exception("No valid schema matching model for column '"+target+"' in lens '"+viewName+"'");
+      }
+    })
+
+
+    val (
+      schemaChoice:List[(String,Expression)], 
+      metaModels:List[Model]
+    ) =
+      candidateModels.
+        toList.
+        map({ 
+          case (column, models) => {
+            //TODO: Replace Default Model
+            val metaModel = new DefaultMetaModel(
+                s"$viewName:META:$column", 
+                s"picking a source for column '$column'",
+                models.map(_._4)
+              )
+            val metaExpr = LensUtils.buildMetaModel(
+              metaModel.name, 0, Seq[Expression](), Seq[Expression](),
+              models, Seq[Expression]()
+            )
+
+            ( (column, metaExpr), metaModel )
+          }
+        }).
+        unzip
+
+    val proxyModel = SchemaMatchingProxyModel(s"$viewName:PROXY:SCHEMA_MATCHING", targetSchema, targetSchema.map(sche => modelEntities(modelEntities.indexWhere(_.name.endsWith(s":${sche._1}")) ).name ))   
+    modelEntities ++ metaModels :+ proxyModel
+  }
+
+  def tableCatalogFor(db: Database, config: MultilensConfig): Operator =
+  {
+    HardTable(
+      Seq(
+        ("TABLE_NAME",TString())
+      ),
+      Seq(
+        Seq(
+          StringPrimitive("DATA")
+        )
+      )
+    )
+  }
+  
+  def attrCatalogFor(db: Database, config: MultilensConfig): Operator =
+  {
+    HardTable(
+      Seq(
+        ("TABLE_NAME" , TString()), 
+        ("ATTR_TYPE", TType()),
+        ("IS_KEY", TBool()),
+        ("COLUMN_ID", TInt())
+      ),
+      (0 until config.args.size).map { col =>
+        Seq(
+          StringPrimitive("DATA"),
+          TypePrimitive(TString()),
+          BoolPrimitive(false),
+          IntPrimitive(col)
+        )
+      }
+    ).addColumn(
+      "ATTR_NAME" -> VGTerm(s"${config.schema}:PROXY:SCHEMA_MATCHING" , 1, Seq(Var("COLUMN_ID")), Seq())
+    ).removeColumn("COLUMN_ID")
+  }
+        
+  def viewFor(db: Database, config: MultilensConfig, table: String): Option[Operator] =
+  {
+    if(table.equals("DATA")){
+      val model = db.models.get(s"${config.schema}:PROXY:SCHEMA_MATCHING").asInstanceOf[SchemaMatchingProxyModel]
```

Comment:
Why not invoke the meta models (`s"${config.schema}:META:$column"`) directly?

Reply:
oh... because you need to flatten the multiple meta-models out into a single model so that it works with `attrCatalogFor`

1. This deserves a comment up above when the PROXY model is first defined.
2. This model stacking makes for a nice separation of concerns, but it adds another layer of model look-ups.  Is there a middle ground?

Reply:
I will comment the proxy model before merging to master.  I am no longer using meta models for schema matching - the proxy model flattens the schema matching models directly, so there is only one layer of stacking.  I may need to revisit this before merging.

Reply:
That works